[Part 1] All numeric CSSPrimitiveValue resolvers need to take CSSToLengthConversionData

weinig opened the pull request:
#### b59ab1a6d619275fa32c3ee570f7b4cf0d97cd3d
<pre>
[Part 1] All numeric CSSPrimitiveValue resolvers need to take CSSToLengthConversionData
<a href="https://bugs.webkit.org/show_bug.cgi?id=278753">https://bugs.webkit.org/show_bug.cgi?id=278753</a>

Reviewed by Darin Adler.

A quirk of `calc()` that the engine does not currently account for is that
any numeric CSS value (&lt;number&gt;, &lt;percentage&gt;, &lt;angle&gt;, etc.) can rely on
relative length computations via calc(). For example animation-delay, which
is specified as taking &lt;time&gt; should support the following:

    `animation-delay: calc(10s + (sign(2cqw - 10px) * 5s));`

This can evaluate to either `15s`, if the sign() expression finds a positive
value, or `5s` if it does not.

The ramification of this is that all resolution of CSSPrimitiveValues to a
number require passing in CSSToLengthConversionData, not just when it is
a &lt;length&gt; value.

Making that change everywhere is going to take a few passes. This first change
rearranges CSSPrimitiveValue with a slew of numeric type specific resolver
functions, bundled together in sets with:
    - a normal one, taking a CSSToLengthConversionData
    - one with the suffix `NoConversionDataRequired`, which should only be
      used when the caller has determined ahead of time it won&apos;t be needed,
      such as when `calc()` has already been fully ruled out.
    - one with the suffix `Deprecated`,  which should NOT be used, and is
      only around during the transition.

To make things a bit more clear, instead of passing in a CSSUnitType to
the resolver functions, type specific functions are provided. The old
mechanism was too general, with no callers ever wanting anything but
values of the same category and for the most part, in the canonical unit.

Now, instead of calling:

    `primitiveValue-&gt;floatValue(CSSUnitType::CSS_PERCENTAGE)`

users now call:

     `primitiveValue-&gt;resolveAsPercentage&lt;float&gt;(conversionData)`

Most callers were converted to their `Deprecated` equivalent, but where
it was trivially to pass in CSSToLengthConversionData, the real version
has been used.

* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-delay-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-delay-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-duration-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-duration-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-iteration-count-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-iteration-count-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/background-image-computed.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/border-image-outset-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/border-image-outset-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/border-image-slice-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/border-image-slice-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-feature-settings-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-feature-settings-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/animation/image-slice-interpolation-math-functions-tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-inline/parsing/line-height-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-inline/parsing/line-height-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-increment-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-increment-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-reset-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-reset-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-set-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-set-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-syntax-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-syntax-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/tab-size-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/tab-size-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/parsing/zoom-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/parsing/zoom-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/animation/offset-rotate-interpolation-math-functions-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/animation/ray-angle-interpolation-math-functions-expected.txt:
    - Adds / updates tests to include cases where relative lengths are used
      in non-&lt;length&gt; values.

* Source/WebCore/animation/ViewTimeline.cpp:
* Source/WebCore/css/CSSBasicShapes.cpp:
* Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
* Source/WebCore/css/CSSCrossfadeValue.cpp:
* Source/WebCore/css/CSSFontFace.cpp:
* Source/WebCore/css/CSSFontFeatureValue.cpp:
* Source/WebCore/css/CSSFontFeatureValue.h:
* Source/WebCore/css/CSSGradientValue.cpp:
* Source/WebCore/css/CSSImageSetValue.cpp:
* Source/WebCore/css/CSSOffsetRotateValue.cpp:
* Source/WebCore/css/CSSPrimitiveValue.cpp:
* Source/WebCore/css/CSSPrimitiveValue.h:
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/CSSShapeSegmentValue.cpp:
* Source/WebCore/css/CSSToStyleMap.cpp:
* Source/WebCore/css/CSSToStyleMap.h:
* Source/WebCore/css/CSSValue.h:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
* Source/WebCore/css/DeprecatedCSSOMPrimitiveValue.cpp:
* Source/WebCore/css/SVGCSSComputedStyleDeclaration.cpp:
* Source/WebCore/css/ShorthandSerializer.cpp:
* Source/WebCore/css/StyleProperties.cpp:
* Source/WebCore/css/TransformFunctions.cpp:
* Source/WebCore/css/TransformFunctions.h:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/css/calc/CSSCalcValue.cpp:
* Source/WebCore/css/calc/CSSCalcValue.h:
* Source/WebCore/css/parser/CSSParserImpl.cpp:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/query/GenericMediaQueryEvaluator.cpp:
* Source/WebCore/css/query/GenericMediaQueryParser.cpp:
* Source/WebCore/css/typedom/CSSStyleValueFactory.cpp:
* Source/WebCore/editing/ApplyStyleCommand.cpp:
* Source/WebCore/editing/EditingStyle.cpp:
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
* Source/WebCore/page/IntersectionObserver.cpp:
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
* Source/WebCore/style/FilterOperationsBuilder.cpp:
* Source/WebCore/style/StyleBuilder.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleBuilderState.cpp:
* Source/WebCore/svg/SVGLengthValue.cpp:
    - Update to call new primitive value functions.

Canonical link: <a href="https://commits.webkit.org/282914@main">https://commits.webkit.org/282914@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c4e98523ecb9569d99d68b9e588fa537d58218e6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/64651 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/44016 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/17247 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/68674 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/15257 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/66768 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/51787 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/15536 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/51998 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/10531 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/67717 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/40731 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55944 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32620 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37396 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/13319 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/14130 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/59324 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/13649 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/70379 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/8596 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/13151 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/59330 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/8630 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/56029 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/59510 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14263 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/7109 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/777 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/39828 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/40905 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/42088 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/40649 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on LayoutTests/fast/css/image-set-parsing.html:
```diff
@@ -75,12 +75,12 @@
                 "-webkit-mask-box-image-source",
                 "-webkit-cross-fade(url(\"#a\"), url(\"#b\"), 70%)",
-                "-webkit-cross-fade(url(\"#a\"), url(\"#b\"), 0.7) 1x");
+                "-webkit-cross-fade(url(\"#a\"), url(\"#b\"), 70%) 1x");
```
Is this test change needed because of a behavior change outside of better `calc()` handling? Should we factor that behavior change out into a much smaller pull request that we can merge first?

darinadler commented on Source/WebCore/css/CSSFontFeatureValue.cpp:
```diff
@@ -45,15 +45,15 @@ String CSSFontFeatureValue::customCSSText() const
-    if (m_value != 1)
-        builder.append(' ', m_value);
+    // Omit the value if it's `1` as `1` is implied by default.
+    if (m_value->isCalculated() || m_value->resolveIntegerNoConversionDataRequired() != 1)
```
Should we have something that returns an `optional` instead of using `isCalculated` as a preflight? It’s not obvious that `isCalculated` and `NoConversionDataRequired` are a perfect match. Doesn’t seem that unusual to want a value excluding the calculated one.

weinig replied:
I definitely want to do this, but doing it right now is a bit of a pull. Going to do it as part of the next stage.

darinadler commented on Source/WebCore/css/CSSGradientValue.cpp:
```diff
@@ -95,24 +95,22 @@ template<StopPositionResolution resolution> static inline std::optional<Length>
 
     if (position->isAngle())
-        return { AngleRaw { position->primitiveType(), position->doubleValue() } };
+        return { AngleRaw { CSSUnitType::CSS_DEG, position->resolveAsAngle(state.cssToLengthConversionData()) } };
```
Since `CSS_DEG` here is explicit, maybe the template parameter specifying degrees as the unit should be as well? It would be inelegant but clearer.

darinadler commented on Source/WebCore/css/CSSPrimitiveValue.cpp:
```diff
@@ -615,52 +615,47 @@ Ref<CSSPrimitiveValue> CSSPrimitiveValue::createURI(String value)
+template<> float CSSPrimitiveValue::resolveAsLength(const CSSToLengthConversionData& conversionData) const
 {
-    return roundForImpreciseConversion<unsigned short>(computeLengthDouble(conversionData));
+    return static_cast<float>(resolveAsLengthDouble(conversionData));
```
Does the `static_cast` here do anything? What would happen with no cast at all? Do we still have `narrowPrecisionToFloat` and when would we use that instead?

weinig replied:
Went with `narrowPrecisionToFloat`.

darinadler commented on Source/WebCore/css/CSSToStyleMap.cpp:
```diff
@@ -537,13 +537,19 @@ void CSSToStyleMap::mapNinePieceImageSlice(const CSSValue& value, NinePieceImage
 {
     // Set up a length box to represent our image slices.
-    auto side = [](const CSSPrimitiveValue& value) -> Length {
+    auto side = [](const CSSPrimitiveValue& value, const CSSToLengthConversionData& conversionData) -> Length {
```
Why not capture the conversion data instead of passing it, to keep the `setImageSlices` call easy to read?

darinadler commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
```diff
@@ -1223,17 +1223,19 @@ RefPtr<CSSValue> consumeFeatureTagValue(CSSParserTokenRange& range)
         return nullptr;
 
-    int tagValue = 1;
+    RefPtr<CSSPrimitiveValue> tagValue = nullptr;
```
Omit the `= nullptr`?

darinadler commented on Source/WebCore/css/CSSPrimitiveValue.h:
```diff
@@ -155,42 +157,66 @@ class CSSPrimitiveValue final : public CSSValue {
+    ExceptionOr<float> getFloatValueDeprecated(CSSUnitType targetUnit) const;
+
+    // MARK: Integer (requires `isInteger() == true`)
+    template<typename T = int> T resolveInteger(const CSSToLengthConversionData&) const;
```
Not sure why this name doesn’t include `As`.

weinig replied:
Oops. Nope, just a goof. Fixed.

darinadler commented on Source/WebCore/css/CSSPrimitiveValue.h:
```diff
@@ -155,42 +157,66 @@ class CSSPrimitiveValue final : public CSSValue {
+    template<typename T = int> T resolveIntegerNoConversionDataRequired() const;
+    template<typename T = int> T resolveIntegerDeprecated() const;
+
+    // MARK: Number (requires `isNumberOrInteger() == true`)
```
Unclear that `isNumberOrInteger` corresponds to `AsNumber`. The latter sounds like it could do other conversions.

weinig replied:
There is no conversion here, numbers and integers are both stored as `double`. I need to think on this though, as there may be no need for the `OrInteger`.

weinig committed:
[Part 1] All numeric CSSPrimitiveValue resolvers need to take CSSToLengthConversionData
https://bugs.webkit.org/show_bug.cgi?id=278753

Reviewed by Darin Adler.

A quirk of `calc()` that the engine does not currently account for is that
any numeric CSS value (<number>, <percentage>, <angle>, etc.) can rely on
relative length computations via calc(). For example animation-delay, which
is specified as taking <time> should support the following:

    `animation-delay: calc(10s + (sign(2cqw - 10px) * 5s));`

This can evaluate to either `15s`, if the sign() expression finds a positive
value, or `5s` if it does not.

The ramification of this is that all resolution of CSSPrimitiveValues to a
number require passing in CSSToLengthConversionData, not just when it is
a <length> value.

Making that change everywhere is going to take a few passes. This first change
rearranges CSSPrimitiveValue with a slew of numeric type specific resolver
functions, bundled together in sets with:
    - a normal one, taking a CSSToLengthConversionData
    - one with the suffix `NoConversionDataRequired`, which should only be
      used when the caller has determined ahead of time it won't be needed,
      such as when `calc()` has already been fully ruled out.
    - one with the suffix `Deprecated`,  which should NOT be used, and is
      only around during the transition.

To make things a bit more clear, instead of passing in a CSSUnitType to
the resolver functions, type specific functions are provided. The old
mechanism was too general, with no callers ever wanting anything but
values of the same category and for the most part, in the canonical unit.

Now, instead of calling:

    `primitiveValue->floatValue(CSSUnitType::CSS_PERCENTAGE)`

users now call:

     `primitiveValue->resolveAsPercentage<float>(conversionData)`

Most callers were converted to their `Deprecated` equivalent, but where
it was trivially to pass in CSSToLengthConversionData, the real version
has been used.

* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-delay-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-delay-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-duration-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-duration-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-iteration-count-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-animations/parsing/animation-iteration-count-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/background-image-computed.sub.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/border-image-outset-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/border-image-outset-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/border-image-slice-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/border-image-slice-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-feature-settings-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-feature-settings-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/animation/image-slice-interpolation-math-functions-tentative-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-inline/parsing/line-height-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-inline/parsing/line-height-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-increment-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-increment-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-reset-computed-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-reset-computed.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-set-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-lists/parsing/counter-set-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-syntax-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-properties-values-api/register-property-syntax-parsing.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/tab-size-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-text/parsing/tab-size-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/parsing/zoom-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/parsing/zoom-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/animation/offset-rotate-interpolation-math-functions-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/motion/animation/ray-angle-interpolation-math-functions-expected.txt:
    - Adds / updates tests to include cases where relative lengths are used
      in non-<length> values.

* Source/WebCore/animation/ViewTimeline.cpp:
* Source/WebCore/css/CSSBasicShapes.cpp:
* Source/WebCore/css/CSSCounterStyleDescriptors.cpp:
* Source/WebCore/css/CSSCrossfadeValue.cpp:
* Source/WebCore/css/CSSFontFace.cpp:
* Source/WebCore/css/CSSFontFeatureValue.cpp:
* Source/WebCore/css/CSSFontFeatureValue.h:
* Source/WebCore/css/CSSGradientValue.cpp:
* Source/WebCore/css/CSSImageSetValue.cpp:
* Source/WebCore/css/CSSOffsetRotateValue.cpp:
* Source/WebCore/css/CSSPrimitiveValue.cpp:
* Source/WebCore/css/CSSPrimitiveValue.h:
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/CSSShapeSegmentValue.cpp:
* Source/WebCore/css/CSSToStyleMap.cpp:
* Source/WebCore/css/CSSToStyleMap.h:
* Source/WebCore/css/CSSValue.h:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
* Source/WebCore/css/DeprecatedCSSOMPrimitiveValue.cpp:
* Source/WebCore/css/SVGCSSComputedStyleDeclaration.cpp:
* Source/WebCore/css/ShorthandSerializer.cpp:
* Source/WebCore/css/StyleProperties.cpp:
* Source/WebCore/css/TransformFunctions.cpp:
* Source/WebCore/css/TransformFunctions.h:
* Source/WebCore/css/calc/CSSCalcTree.h:
* Source/WebCore/css/calc/CSSCalcValue.cpp:
* Source/WebCore/css/calc/CSSCalcValue.h:
* Source/WebCore/css/parser/CSSParserImpl.cpp:
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp:
* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
* Source/WebCore/css/query/GenericMediaQueryEvaluator.cpp:
* Source/WebCore/css/query/GenericMediaQueryParser.cpp:
* Source/WebCore/css/typedom/CSSStyleValueFactory.cpp:
* Source/WebCore/editing/ApplyStyleCommand.cpp:
* Source/WebCore/editing/EditingStyle.cpp:
* Source/WebCore/editing/cocoa/HTMLConverter.mm:
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
* Source/WebCore/page/IntersectionObserver.cpp:
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
* Source/WebCore/style/FilterOperationsBuilder.cpp:
* Source/WebCore/style/StyleBuilder.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleBuilderState.cpp:
* Source/WebCore/svg/SVGLengthValue.cpp:
    - Update to call new primitive value functions.

Canonical link: https://commits.webkit.org/282914@main
